sdk/core/core/src/az_span_private.h

```diff
@@ -35,6 +35,9 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index);
 
 AZ_NODISCARD az_result _az_is_expected_span(az_span* self, az_span expected);
 
+// Used internally by json parser and also by httpRequest
+AZ_NODISCARD az_result az_span_reader_skip_json_white_space(az_span* self);
```

Comment:
Rename this since it is no longer json specific.

I think something like this could work:
```C
AZ_NODISCARD az_span az_span_trim_white_space(az_span span)
AZ_NODISCARD az_span az_span_trim_white_space_from_start(az_span span)
AZ_NODISCARD az_span az_span_trim_white_space_from_end(az_span span)
```

Reply:
Yes, I like Ahson's signatures. And yes, we can have them be private.

Reply:
Done